Add a guide for Caddy

jensz12 opened the pull request:
Added a guide for how to handle reverse proxy using Caddy. Basically just a copy of the Nginx guide

TBlueF commented:
Would it be possible to merge this with the nginx page instead and make it a general "Reverse-Proxy" page instead?
To keep it a bit more organized :)

TechnicJelle commented:
Would it be possible to keep the current URL `https://bluemap.bluecolored.de/wiki/webserver/NginxProxy.html` working? Perhaps via a redirect?

TechnicJelle commented:
The combining could be done in the same style as https://bluemap.bluecolored.de/wiki/webserver/ExternalWebserversFile.html and https://bluemap.bluecolored.de/wiki/webserver/ExternalWebserversSQL.html are

jensz12 commented:
@TBlueF @TechnicJelle I changed the NGINX page and added the caddy config into it. Im not sure how doing redirects is handled here, so i will leave that to you guys
